### DISCUSSION TITLE
Moderation report routing

### DISCUSSION THREAD

**Discussion by imax9000** [bluesky-social/atproto#2331] [Clarifications and "Why?"]
How does a user report get routed to the moderation service responsible for the target user's content?

**Comment by devinivy** [comment]
It uses a new PDS functionality called [service proxying](https://atproto.com/specs/xrpc#service-proxying).  The client makes a `com.atproto.moderation.createReport` request to the user's pds with the `atproto-proxy` header set to reference the service that the request should be proxied to.  The pds then makes this request with a signed JWT which authenticates the user with that service.

**Reply by imax9000** [reply]
1) so it's a client's obligation to determine which moderation service corresponds to target's PDS?
2) how does the client find out the relevant endpoint starting from a post/profile to be reported?
